### DISCUSSION TITLE
Is there a way to specify distinct requires-python for development and for releases of a project that is release as a pypi library?

### DISCUSSION THREAD

**Discussion by fgmacedo** [pdm-project/pdm#2997] [Q&A] [answered]
I'm the author of a library called [python-statemachine](https://github.com/fgmacedo/python-statemachine).

The library only optionally depends on `pydot`, and the lib works on Python3.7+.

But I'm suffering to keep working on the project in development mode if I set `requires-python = ">=3.7"`, It stops resolving dependencies even if they were on dev/test groups, for example, `myst-parser` requires Python>=3.8.

I develop on Python 3.12, and have a test matrix from 3.7..3.12 on GitHub as CI.

Is it possible to get this? 

Best!

PS: I currently use Poetry and considering other alternatives as I cannot figure out how to accomplish this.

**Comment by frostming** [comment, marked as answer]
if you make sure some dependencies will only be used on Python 3.8+, restrict it with environment markers:

```
myst-parser; python_version>="3.8"
```

and it can be resolved correctly.

also, the new feature #2995 to be landed in the next release will also help in this case
